Replace extract-zip

This package is using a fork.

It's using a pretty outdated version of `debug`, but is has a vulnerable version of `ms`. Open PRs on the repo have been open for a while, and the tests do not pass on Mac (though, the failure is only related to error messages).

Can't currently install kbn-shoehorn without access to the fork:

```
npm info it worked if it ends with ok
npm info using npm@3.10.10
npm info using node@v6.10.3
npm info addNameTag [ 'kbn-shoehorn', 'latest' ]
npm info attempt registry request try #1 at 11:43:39 PM
npm http request GET https://registry.npmjs.org/@w33ble%2fextract-zip
npm http 404 https://registry.npmjs.org/@w33ble%2fextract-zip
npm ERR! Linux 4.4.59-1-pve
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "-g" "install" "kbn-shoehorn" "-d"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code E404

npm ERR! 404 Not found : @w33ble/extract-zip
npm ERR! 404 
npm ERR! 404  '@w33ble/extract-zip' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'kbn-shoehorn'
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

--- a/package.json
+++ b/package.json
@@ -22,12 +22,13 @@
   },
   "homepage": "https://github.com/w33ble/kbn-shoehorn",
   "dependencies": {
-    "@w33ble/extract-zip": "^1.6.6",
     "download-git-repo": "^1.0.1",
     "inquirer": "^3.2.0",
     "minimist": "^1.2.0",
+    "mkdirp": "^0.5.1",
     "ncp": "^2.0.0",
-    "temp-dir": "^1.0.0"
+    "temp-dir": "^1.0.0",
+    "yauzl": "^2.8.0"
   },
   "devDependencies": {
     "eslint": "^3.19.0",

--- a/src/extract.js
+++ b/src/extract.js
@@ -1,0 +1,75 @@
+/*
+  Heavily aped from https://github.com/elastic/kibana/blob/5.6/src/cli_plugin/install/zip.js
+
+  Modified for local eslint rules and fancy ES6 syntax
+*/
+
+const { createWriteStream } = require('fs');
+const { join, dirname } = require('path');
+const yauzl = require('yauzl');
+const mkdirp = require('mkdirp');
+
+const isDirectory = filename => /(\/|\\)$/.test(filename);
+
+const createPath = p => new Promise((resolve, reject) => {
+  mkdirp(p, (err) => {
+    if (err) reject(err);
+    else resolve(p);
+  });
+});
+
+const openZip = (zip, opts = {}) => new Promise((resolve, reject) => {
+  yauzl.open(zip, opts, (err, zipfile) => {
+    if (err) reject(err);
+    else resolve(zipfile);
+  });
+});
+
+const writeEntity = (zipfile, entity, fileName) => {
+  return new Promise((resolve, reject) => {
+    // write the file using the streaming output interface
+    zipfile.openReadStream(entity, (err, readStream) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      // ensure parent directory exists
+      createPath(dirname(fileName)).then(() => {
+        // write the target file
+        readStream.pipe(createWriteStream(fileName));
+
+        // on completion, process next file in zip
+        readStream.on('end', resolve());
+      }).catch(e => reject(e));
+    });
+  });
+};
+
+module.exports = function extractArchive(zip, opts) {
+  const targetDir = opts.dir;
+  const zipOpts = { lazyEntries: opts.lazyEntries || true };
+
+  if (!targetDir) return Promise.reject('No target dir specified for unzippping');
+
+  return new Promise((resolve, reject) => {
+    openZip(zip, zipOpts).then((zipfile) => {
+      const processNextFile = () => zipfile.readEntry();
+
+      zipfile.on('close', resolve);
+
+      zipfile.on('entry', (entry) => {
+        const fileName = join(targetDir, entry.fileName);
+
+        const chain = isDirectory(fileName)
+          ? createPath(fileName)
+          : writeEntity(zipfile, entry, fileName);
+
+        chain.then(processNextFile).catch(err => reject(err));
+      });
+
+      // kick off the zip file processing
+      processNextFile();
+    });
+  });
+};

--- a/src/json_file.js
+++ b/src/json_file.js
@@ -1,9 +1,5 @@
 const { readFileSync, writeFileSync } = require('fs');
 
-exports.readJson = (path) => {
-  return JSON.parse(readFileSync(path));
-};
+exports.readJson = path => JSON.parse(readFileSync(path));
 
-exports.writeJson = (path, content) => {
-  return writeFileSync(path, JSON.stringify(content, null, 2));
-};
+exports.writeJson = (path, content) => writeFileSync(path, JSON.stringify(content, null, 2));

--- a/src/load_file.js
+++ b/src/load_file.js
@@ -8,6 +8,7 @@ const { readJson } = require('./json_file');
 // paths to ignore when attempting to traverse into plugin
 const ignorePaths = [
   '.DS_Store',
+  '__MACOSX',
 ];
 
 function readPackageName(dirPath) {

--- a/src/load_file.js
+++ b/src/load_file.js
@@ -2,7 +2,7 @@ const { statSync, readdirSync } = require('fs');
 const { parse, join } = require('path');
 const ncp = require('ncp').ncp;
 const tempDir = require('temp-dir');
-const extract = require('@w33ble/extract-zip');
+const extract = require('./extract');
 const { readJson } = require('./json_file');
 
 // paths to ignore when attempting to traverse into plugin
@@ -39,41 +39,36 @@ function copyPath(filePath, targetPath) {
 }
 
 function unpackZip(filePath, targetPath) {
-  return new Promise((resolve, reject) => {
-    const { ext, name } = parse(filePath);
-    const unpackTarget = join(tempDir, name, String((new Date()).getTime())); // unique temp unzip path
+  const { ext, name } = parse(filePath);
+  // unique temp unzip path
+  const unpackTarget = join(tempDir, name, String((new Date()).getTime()));
+  console.log('unpackTarget', unpackTarget)
 
-    if (ext !== '.zip') {
-      reject(`Plugin must be a .zip file, ${ext} is not supported`);
-      return;
+  if (ext !== '.zip') {
+    return Promise.reject(`Plugin must be a .zip file, ${ext} is not supported`);
+  }
+
+  return extract(filePath, { dir: unpackTarget })
+  .then(() => {
+    // look for name in package.json, traverse if only one path is found
+    function getPluginPath(dirPath) {
+      const pluginName = readPackageName(dirPath);
+
+      if (!pluginName) {
+        const paths = readdirSync(dirPath).filter(path => ignorePaths.indexOf(path) === -1);
+        if (paths.length !== 1) throw new Error(`Plugin not found: ${filePath}`);
+
+        const deepDirPath = join(dirPath, paths[0]);
+        const stat = statSync(deepDirPath);
+        if (!stat.isDirectory()) throw new Error(`Plugin not found: ${filePath}`);
+
+        return getPluginPath(deepDirPath);
+      }
+
+      return dirPath;
     }
 
-    extract(filePath, { dir: unpackTarget }, (err) => {
-      if (err) {
-        reject(err);
-        return;
-      }
-
-      // look for name in package.json, traverse if only one path is found
-      function getPluginPath(dirPath) {
-        const pluginName = readPackageName(dirPath);
-
-        if (!pluginName) {
-          const paths = readdirSync(dirPath).filter(path => ignorePaths.indexOf(path) === -1);
-          if (paths.length !== 1) throw new Error(`Plugin not found: ${filePath}`);
-
-          const deepDirPath = join(dirPath, paths[0]);
-          const stat = statSync(deepDirPath);
-          if (!stat.isDirectory()) throw new Error(`Plugin not found: ${filePath}`);
-
-          return getPluginPath(deepDirPath);
-        }
-
-        return dirPath;
-      }
-
-      resolve(copyPath(getPluginPath(unpackTarget), targetPath));
-    });
+    return copyPath(getPluginPath(unpackTarget), targetPath);
   });
 }
 

